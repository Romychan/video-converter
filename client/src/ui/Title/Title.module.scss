.container {
  display: flex;
  align-items: center;
  width: 100%;
}

.link {
  display: flex;
  align-items: center;
}

.icon {
  margin-right: 8px;
  stroke: var(--text-color-primary);
  stroke-width: 1.8px;
  transition: 0.2s;
}

.title {
}
