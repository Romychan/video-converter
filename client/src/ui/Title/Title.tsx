import clsx from 'clsx';
import { Link } from 'react-router-dom';

import { Icon } from '../Icon/Icon';

import styles from './Title.module.scss';

interface ITitleProps {
  link: string;
  title: string;
  className?: string;
}

export const Title = ({ link, title, className = '' }: ITitleProps) => {
  return (
    <div className={clsx(styles.container, className)}>
      <Link className={styles.link} to={link} data-testid="title-link">
        <Icon name="arrow-back" size={21} className={styles.icon} />
        <h1 className={clsx(styles.title, 'title')}>{title}</h1>
      </Link>
    </div>
  );
};
