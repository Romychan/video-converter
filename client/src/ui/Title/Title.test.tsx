import { screen } from '@testing-library/react';

import { renderWithRouter } from '@tests/helpers/renderWithRouter';
import { Title } from './Title';

describe('Title', () => {
  it('should render correctly', () => {
    renderWithRouter(<Title link="/user" title="Testing" />);

    expect(screen.getByText(/Testing/i)).toBeInTheDocument();
    expect(screen.getByTestId('title-link')).toHaveAttribute('href', '/user');
  });

  it('should match snapshot', () => {
    const container = renderWithRouter(<Title link="/user" title="Testing" />);
    expect(container).toMatchSnapshot();
  });
});
