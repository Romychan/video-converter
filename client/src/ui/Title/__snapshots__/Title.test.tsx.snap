// Vitest Snapshot v1

exports[`Title > should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_container_4b116b"
      >
        <a
          class="_link_4b116b"
          data-testid="title-link"
          href="/user"
        >
          <svg
            class="icon icon-arrow-back _icon_4b116b"
            data-testid="icon"
            height="21"
            stroke=""
            width="21"
          >
            <use
              xlink:href="/src/assets/img/general/icons.svg#arrow-back"
            />
          </svg>
          <h1
            class="_title_4b116b title"
          >
            Testing
          </h1>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_container_4b116b"
    >
      <a
        class="_link_4b116b"
        data-testid="title-link"
        href="/user"
      >
        <svg
          class="icon icon-arrow-back _icon_4b116b"
          data-testid="icon"
          height="21"
          stroke=""
          width="21"
        >
          <use
            xlink:href="/src/assets/img/general/icons.svg#arrow-back"
          />
        </svg>
        <h1
          class="_title_4b116b title"
        >
          Testing
        </h1>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
